projects/WhatsAppClone/src/components/channel/MessageInput.tsx

```diff
@@ -0,0 +1,187 @@
+    'state',
+    'members',
+  ]) as ChannelState<StreamChatGenerics>['members']
+  const numberOfMembers = size(channelMembers)
+  const recipientChannelMemberName = get(values(channelMembers), [
+    1,
+    'user',
+    'name',
+  ])
+
+  const isMessageEmpty = useMemo(
+    () => !text && !imageUploads.length && !fileUploads.length,
+    [text, imageUploads.length, fileUploads.length],
+  )
+
+  const takeAndUploadImage = useCallback(async () => {
+    setSelectedPicker(undefined)
+    closePicker()
+    const photo = await takePhoto({compressImageQuality})
+    if (!photo.cancelled) {
+      await uploadNewImage(photo)
+      navigate('ImagePreview')
+    }
+  }, [closePicker, compressImageQuality, setSelectedImages, setSelectedPicker])
+
+  useEffect(() => {
+    if (numberOfFiles === 0) return
+    const allFilesAreReady = fileUploads.every(f => !isEmpty(f.url))
+
+    if (!allFilesAreReady) return
+
+    const fileUploadsDescription =
+      numberOfFiles > 1
+        ? `${numberOfFiles} files`
+        : `'${get(fileUploads, [0, 'file', 'name'])}'`
+
+    const membersDescription =
+      numberOfMembers > 2
+        ? `all ${numberOfMembers} members`
+        : `${recipientChannelMemberName}`
+
+    Alert.alert(
+      'Send file/s',
+      `Send ${fileUploadsDescription} to ${membersDescription}?`,
+      [
+        {onPress: () => null, text: 'Cancel'},
+        {
+          onPress: sendMessage,
+          text: 'Send',
+        },
+      ],
+      {cancelable: true},
+    )
+  }, [JSON.stringify(fileUploads)])
+
+  const isReplyPreviewEnabled = Boolean(
+    (typeof editing !== 'boolean' && editing?.quoted_message) || quotedMessage,
+  )
+
+  const formattedAudioDuration = useMemo(
+    () => moment(recordingDurationInMS).format('m:ss'),
+    [recordingDurationInMS],
+  )
+
+  return (
+    <SafeAreaView style={styles.outerContainer}>
+      <View
+        style={{
+          ...styles.innerContainer,
+          ...(isReplyPreviewEnabled
+            ? {borderTopRightRadius: sizes.ml, borderTopLeftRadius: sizes.ml}
+            : {}),
+        }}>
+        <Reply
+          isPreview
+          isEnabled={isReplyPreviewEnabled}
+          message={quotedMessage}
+        />
+        <View style={flex.directionRowItemsCenter}>
+          <PeekabooView isEnabled={recordingActive}>
+            <RecordingBlinking />
+            <View style={{flex: 1}}>
+              <Text style={{color: colors.dark.secondaryLight}}>
+                {formattedAudioDuration}
+              </Text>
+            </View>
+          </PeekabooView>
+          <PeekabooView isEnabled={!recordingActive}>
+            <IconButton
+              onPress={() => null}
+              iconName={'Smiley'}
+              pathFill={colors.dark.secondaryLight}
+            />
+            <AutoCompleteInput
+              setInputBoxRef={
+                messageInputRef as unknown as LocalMessageInputContext['setInputBoxRef']
```

Comment:
Why is this needed ? 🤔

Reply:
Our `AutoCompleteInput` expects `setInputBoxRef: (ref: TextInput | null) => void;`

Where my local context messageInputRef is `RefObject<TextInput>` originating from a regular `useRef` hook `const messageInputRef = useRef<TextInput>(null)`

Any ideas on how to solve this?

Reply:
`useRef` will always wrap `RefObject` 🤔

Reply:
This should be wrapped in a RefOvject in our SDK, improve typing- create issue